Sources/Workspace/Workspace.swift

```diff
@@ -941,6 +947,17 @@ public class Workspace {
                 case .checkout(let checkoutState):
                     packagePath = checkoutsPath.appending(managedDependency.subpath)
                     version = checkoutState.version
+                    
+                    // In case the checkouts folder for the dependency was deleted,
+                    // clone it again.
+                    if !fileSystem.isDirectory(packagePath) {
```

Comment:
Instead of doing this here, I suggest renaming `validateEditedPackages` to `validateManagedDependencies` and check for validity of both edited as well as managed checkouts.